lib/data/managers/firestore_manager.dart

```diff
@@ -19,64 +19,61 @@ class FireStoreManager {
   FireStoreManager._internal();
 
   Future<List<CropEntity>> getCrops() async {
-    List<CropEntity> cropsEntities;
-
     // Filters defined by product definition. - CROPS
     var query = Firestore.instance
         .collection(FLAME_LINK_CONTENT)
         .where(FLAME_LINK_SCHEMA, isEqualTo: Schema.CROP)
-        .where(FLAME_LINK_ENVIROMENT,
-            isEqualTo: AppSettings.get().environment)
+        .where(FLAME_LINK_ENVIROMENT, isEqualTo: AppSettings.get().environment)
         .where(FLAME_LINK_LOCALE, isEqualTo: Locale.EN_US)
         .where(PUBLICATION_STATUS, isEqualTo: DataStatus.PUBLISHED);
 
-    await query.getDocuments().then((snapshot) {
-      cropsEntities = snapshot.documents.map((cropDocument) {
-        return CropEntity.cropFromDocument(cropDocument);
-      }).toList();
-    });
-    return cropsEntities;
+    return query.getDocuments().then((snapshot) => snapshot.documents
+        .map((cropDocument) => CropEntity.cropFromDocument(cropDocument))
+        .toList());
   }
 
-  Future<List<CropEntity>> getStages(List<CropEntity> cropsList) async {
-    List<CropEntity> cropsWithStages = List();
+  /// Adds stages to the supplied CropEntities
+  Future<dynamic> getStages(List<CropEntity> cropsList) async {
+    List<Future<dynamic>> stageFutures = List();
 
     for (var crop in cropsList) {
       if (crop.stagesPathReference != null) {
         for (var stagesPathReference in crop.stagesPathReference) {
-          await Firestore.instance
```

Comment:
this is the actual point that will improve performance. This allows all fetches for each stage to be done in parallel, instead of waiting for each one.